src/App.tsx

```diff
@@ -1,19 +1,82 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import classnames from 'classnames';
+import { createTodos, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+
+enum Filters {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [value, setValue] = useState('');
+  const [islinkAll, setislinkAll] = useState(true);
+  const [islinkActive, setislinkActive] = useState(false);
+  const [islinkCompleted, setislinkCompleted] = useState(false);
+  const [savedId, setsavedId] = useState(0);
+  const [currentFilter, setCurrentFilter] = useState(Filters.All);
+  const [isError, setisError] = useState(false);
+  const [isHidden, setisHidden] = useState(false);
+  const [completedTodo, setCompletedTodo] = useState<Todo>({
+    id: 0,
+    userId: 0,
+    title: '',
+    completed: false,
+  });
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    try {
+      setisHidden(true);
+      getTodos(user?.id).then(setTodos);
+    } catch (error) {
+      setisError(true);
+    }
+
+    setTimeout(() => {
+      setisHidden(true);
+    }, 3000);
+
+    createTodos(value).then(setTodos);
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const filterList = (): Todo[] | undefined => todos.filter((todo) => {
+    switch (currentFilter) {
+      case Filters.Active:
+        setislinkAll(false);
+        setislinkActive(true);
+        setislinkCompleted(false);
```

Comment:
Are you sure you need all these states? Maybe would be better to create one with an active filter? Or you could use params from the router to get them